### DISCUSSION TITLE
Is there any way to modify the dab config file on publish?

### DISCUSSION THREAD

**Discussion by julianadormon** [Azure/data-api-builder#1738] [Q&A]
Is there any way to modify the dab config file on publish?

**Comment by yorek** [comment]
What do you mean by "publish"?

**Comment by julianadormon** [comment]
My Dab config file is part of my website's project files.  When I publish a staging version, I have a different connection string than in production. I was thinking something kind of like using transforms in web.config? Thanks.

**Reply by yorek** [reply]
In addition to what @Aniruddh25 said, you also have other two options. If you are deploying using Static Web Apps, you're already good to go as you can have different environments, each with its own connection string. (And every PR as a dedicated environment too!): https://learn.microsoft.com/en-us/azure/static-web-apps/preview-environments

If you are not using Static Web Apps, you can use different configuration files by setting the environment variable DAB_CONFIG so that for each environment it will tell DAB to load the correct configuration file. For example: `dab-config.Development.json`: https://learn.microsoft.com/en-us/azure/data-api-builder/configuration-file#environments-support

**Comment by Aniruddh25** [comment]
Hi @julianadormon,

Perhpas, you could use the feature to replace environment variables with their values based on your staging/production version: 
1. use `@env('connection-string')` in the config. 
2. Set the value of `connection-string` to be different for different environments. 
3. One way you could set this is to add different `.env` files in the same directory as the config file for different environments holding different values to the environment variable `connection-string`

For more details, see here: https://learn.microsoft.com/en-us/azure/data-api-builder/configuration-file#setting-environment-variables

**Reply by julianadormon** [reply]
Thank you for this!
